route-rs-runtime/src/link/primitive/classify_link.rs

```diff
@@ -221,7 +208,17 @@ impl<'a, C: Classifier> Future for ClassifyIngressor<'a, C> {
                 ready!(Pin::new(&mut ingressor.input_stream).poll_next(cx));
 
             match packet_option {
-                None => return Poll::Ready(()),
+                None => {
```

Comment:
It seems like each link implements the "eggressor check" in a slightly different way. Is this due to actual different contexts among links, or are they common enough that we can abstract them into the `task_park` module?

Reply:
Hmm, they are the same in the sense that you iterate over the entire collection of egressors, and then die_and_notify for each associated task park. So in this case, fork_link and classify_link have the same flow, and then join_link is its own case, and queue_link is the base case. So, I'm not sure there is much of a common piece we can cut out, without taking additional args into the function. 

I think I'm going to leave the the logic as is, while it isn't the most DRY, it is pretty readable right now.